### PR TITLE
Fix make bundle

### PR DESCRIPTION
**Pull request by rollandf**
- Fix `bundle` target in Makefile
- Rename to nvidia-network-operator
- Run bundle
- Add documentation to test bundle
    
Closes issue #305
    
Signed-off-by: Fred Rolland <frolland@nvidia.com>

### CODE REVIEW

File: config/manifests/bases/nvidia-network-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  keywords:
+  - sriov
+  - networking
+  links:
+  - name: Network Operator Documentation
+    url: https://docs.nvidia.com/networking/display/COKAN10/Network+Operator
+  - name: GitHub repository
+    url: https://github.com/Mellanox/network-operator/
+  maintainers:
+  - email: nvidia-network-operatorsupport@nvidia.com
```
Comment (rollandf) [inline config/manifests/bases/nvidia-network-operator.clusterserviceversion.yaml:60]: @e0ne Is it OK to have this mail here?

File: docs/operator-bundle.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+## Deploying the operator
+
+The operator must be deployed to the nvidia-network-operator namespace. Create the namespace.
+
+```bash
+cat <<EOF | kubectl create -f -
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: nvidia-network-operator
```
Comment (rollandf) [inline docs/operator-bundle.md:47]: @e0ne Should we keep network-operator as namespace?

File: config/manager/kustomization.yaml [hunk 3]
```diff
@@ -12,5 +12,4 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: docker.io/mellanox/network-operator
-  newTag: latest
+  newName: mellanox/network-operator
```
Comment (e0ne) [inline config/manager/kustomization.yaml:15]: Should we refer to NGC repo here?

File: config/manifests/bases/nvidia-network-operator.clusterserviceversion.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+kind: ClusterServiceVersion
+metadata:
+  annotations:
+    alm-examples: ""
+    capabilities: Basic Install
+    description: Deploy and manage NVIDIA networking resources in Kubernetes
+    provider: NVIDIA
+    repository: https://github.com/Mellanox/network-operator/
+    support: NVIDIA
+  name: nvidia-network-operator.v0.0.0
```
Comment (e0ne) [inline config/manifests/bases/nvidia-network-operator.clusterserviceversion.yaml:11]: Why do we have v0.0.0 here? is it a template?

File: config/manifests/bases/nvidia-network-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (e0ne) [reply config/manifests/bases/nvidia-network-operator.clusterserviceversion.yaml:60]: no, it should be: nvidia-network-operator-support@nvidia.com

File: config/manifests/bases/nvidia-network-operator.clusterserviceversion.yaml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  - name: GitHub repository
+    url: https://github.com/Mellanox/network-operator/
+  maintainers:
+  - email: nvidia-network-operatorsupport@nvidia.com
+    name: NVIDIA
+  maturity: alpha
+  provider:
+    name: NVIDIA
+    url: https://github.com/Mellanox/network-operator/
+  version: 0.0.0
```
Comment (e0ne) [inline config/manifests/bases/nvidia-network-operator.clusterserviceversion.yaml:66]: is it a template?

File: docs/operator-bundle.md (hunk 2 above)
Comment (e0ne) [reply docs/operator-bundle.md:47]: I don't have strong opinion here. @adrianchiris any thoughts?

**Review by e0ne — CHANGES_REQUESTED** [review]

File: config/manifests/bases/nvidia-network-operator.clusterserviceversion.yaml (hunk 4 above)
Comment (rollandf) [reply config/manifests/bases/nvidia-network-operator.clusterserviceversion.yaml:11]: Yes, this is the template

File: config/manifests/bases/nvidia-network-operator.clusterserviceversion.yaml (hunk 5 above)
Comment (rollandf) [reply config/manifests/bases/nvidia-network-operator.clusterserviceversion.yaml:66]: Yes

**Comment by AbdYsn** [comment]
kind CI is failing with error:
```
2022-03-13T09:26:23.715Z	ERROR	controller-runtime.source	if kind is a CRD, it should be installed before calling Start	{"kind": "NetworkAttachmentDefinition.k8s.cni.cncf.io", "error": "no matches for kind \"NetworkAttachmentDefinition\" in version \"k8s.cni.cncf.io/v1\""}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132
sigs.k8s.io/controller-runtime/pkg/source.(*Kind).Start
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.1/pkg/source/source.go:117
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.1/pkg/internal/controller/controller.go:166
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.1/pkg/internal/controller/controller.go:222
sigs.k8s.io/controller-runtime/pkg/manager.(*controllerManager).startRunnable.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.1/pkg/manager/internal.go:667
2022-03-13T09:26:23.715Z	INFO	controller-runtime.manager.controller.macvlannetwork	Starting EventSource	{"reconciler group": "mellanox.com", "reconciler kind": "MacvlanNetwork", "source": "kind source: /, Kind="}
2022-03-13T09:26:23.715Z	ERROR	controller-runtime.manager	error received after stop sequence was engaged	{"error": "Timeout: failed waiting for *v1.DaemonSet Informer to sync"}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132
sigs.k8s.io/controller-runtime/pkg/manager.(*controllerManager).engageStopProcedure.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.1/pkg/manager/internal.go:525
I0313 09:26:24.863232       1 request.go:655] Throttling request took 1.04761219s, request: GET:https://172.0.0.1:443/apis/events.k8s.io/v1beta1?timeout=32s
2022-03-13T09:26:25.963Z	ERROR	controller-runtime.source	if kind is a CRD, it should be installed before calling Start	{"kind": "NetworkAttachmentDefinition.k8s.cni.cncf.io", "error": "no matches for kind \"NetworkAttachmentDefinition\" in version \"k8s.cni.cncf.io/v1\""}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132
sigs.k8s.io/controller-runtime/pkg/source.(*Kind).Start
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.1/pkg/source/source.go:117
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.1/pkg/internal/controller/controller.go:166
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.1/pkg/internal/controller/controller.go:222
sigs.k8s.io/controller-runtime/pkg/manager.(*controllerManager).startRunnable.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.1/pkg/manager/internal.go:667
2022-03-13T09:26:25.963Z	ERROR	setup	problem running manager	{"error": "no matches for kind \"NetworkAttachmentDefinition\" in version \"k8s.cni.cncf.io/v1\""}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132
main.main
```
we deploy the operator using the `deploy` make rule

File: config/manifests/bases/nvidia-network-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (rollandf) [reply config/manifests/bases/nvidia-network-operator.clusterserviceversion.yaml]: Fixed

File: config/manager/kustomization.yaml (hunk 3 above)
Comment (rollandf) [reply config/manager/kustomization.yaml]: Updated. It is changed when running `make bundle`

File: docs/operator-bundle.md (hunk 2 above)
Comment (adrianchiris) [reply docs/operator-bundle.md:47]: lets go with nvidia-network-operator.

network-operator is abit too broad IMO

**Comment by rollandf** [comment]
@e0ne CI is green. PTAL when you have time.

File: Makefile [hunk 6]
```diff
@@ -208,28 +209,31 @@ run: generate manifests	## Run against the configured Kubernetes cluster in ~/.k
...
 
 uninstall: manifests	## Uninstall CRDs from a cluster
-	kubectl delete -f config/crd/bases
+	kubectl delete -f deployment/network-operator/crds/
 
 deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
 	cd config/manager && $(KUSTOMIZE) edit set image controller=${TAG}
 	$(KUSTOMIZE) build config/resources-namespace | kubectl apply -f -
 	$(KUSTOMIZE) build config/default | kubectl apply -f -
+	kubectl apply -f hack/crds/*
```
Comment (adrianchiris) [inline Makefile:221]: what is hack/crds ?

File: bundle.Dockerfile [hunk 7]
```diff
@@ -0,0 +1,21 @@
+FROM scratch
+
+# Core bundle labels.
+LABEL operators.operatorframework.io.bundle.mediatype.v1=registry+v1
+LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
+LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
+LABEL operators.operatorframework.io.bundle.package.v1=nvidia-network-operator
+LABEL operators.operatorframework.io.bundle.channels.v1=v1.1.0
+LABEL operators.operatorframework.io.bundle.channel.default.v1=v1.1.0
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.7.1+git
```
Comment (adrianchiris) [inline bundle.Dockerfile:10]: searching commits, i see we migrated to operator-sdk 1.8.0

https://github.com/Mellanox/network-operator/commit/a9c8812bc65d6a2cf93ad345d183e6fc9b3e17ea

does this have any meaning here ?

Comment (rollandf) [reply bundle.Dockerfile:10]: This is just a label, this file is auto-generated by operator-sdk

File: Makefile (hunk 6 above)
Comment (rollandf) [reply Makefile:221]: It contains `networkattachmentdefinitions` CRD

File: Makefile [hunk 8]
```diff
@@ -208,28 +209,31 @@ run: generate manifests	## Run against the configured Kubernetes cluster in ~/.k
 	go run ./main.go
 
 install: manifests	## Install CRDs into a cluster
-	kubectl apply -f config/crd/bases
+	kubectl apply -f deployment/network-operator/crds/
```
[the remark this answers is not in the document]
Comment (rollandf) [reply Makefile:212]: Reverting

File: bundle.Dockerfile (hunk 7 above)
Comment (e0ne) [reply bundle.Dockerfile:10]: @rollandf  which version of operator-sdk did you use?

Comment (rollandf) [reply bundle.Dockerfile:10]:
```
$ operator-sdk version
operator-sdk version: "v1.18.0", commit: "c9c61b6921b29d731e64cd3cc33d268215fb3b25", kubernetes version: "1.21", go version: "go1.17.7", GOOS: "linux", GOARCH: "amd64"
```

**Review by e0ne — APPROVED** [review]
Thanks for addressing my comments

**Review by adrianchiris — APPROVED** [review]

**Pull request merged**